SignalMessaging/attachments/MessageApprovalViewController.swift

```diff
@@ -0,0 +1,231 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc
+public protocol MessageApprovalViewControllerDelegate: class {
+    func messageApproval(_ messageApproval: MessageApprovalViewController, didApproveMessage messageText: String)
+    func messageApprovalDidCancel(_ messageApproval: MessageApprovalViewController)
+}
+
+@objc
+public class MessageApprovalViewController: OWSViewController, UITextViewDelegate {
+
+    let TAG = "[MessageApprovalViewController]"
```

Comment:
I tried to conform to the arrangement of the attachment approval view where possible.